xc/device.ml

```diff
@@ -2034,8 +2034,10 @@ module Dm = struct
     let stop_qemu () = (match (Qemu.pid ~xs domid) with
         | None -> () (* nothing to do *)
         | Some qemu_pid ->
-          if is_upstream_qemu domid then
+          if is_upstream_qemu domid then begin
             QMP_Event.remove domid;
+            xs.Xs.rm (sprintf "/libxl/%d/dm-version" domid)
```

Comment:
I think it should remove the whole `/libxl/<domid>` hierarchy otherwise you still have `/libxl/<domid>` entries left over.

Reply:
Thanks, updated.